### ISSUE TITLE
repairmimetypes is not compatible with SQL backends which are case sensitive

### ISSUE THREAD

**Issue by oparoz** [owncloud/core#14132]
ownCloud uses the `utf8_bin` binary collation with MySQL. It is a case sensitive collation.

If you have a file named DC001.DNG and want to update the media type from `application/octet-stream` to `image/x-dcraw` via the repairmimetypes class, it won't happen because the SQL query will only find files with a `.dng` extension, in lower case.

This case sensitivity problem may affect other parts of ownCloud. I only looked at this class.

One solution for this specific class is to change this
```
return \OC_DB::prepare('
	UPDATE `*PREFIX*filecache`
	SET `mimetype` = (
		SELECT `id`
		FROM `*PREFIX*mimetypes`
		WHERE `mimetype` = ?
	) WHERE `name` LIKE ?
');
```
to this
```
return \OC_DB::prepare('
	UPDATE `*PREFIX*filecache`
	SET `mimetype` = (
		SELECT `id`
		FROM `*PREFIX*mimetypes`
		WHERE `mimetype` = ?
	) WHERE LOWER(`name`) LIKE ?
');
```

**Comment by DeepDiver1975** [comment]
Thanks a lot @oparoz - much appreciated!

Can I ask you to submit a pull request?

Actually we have an ILIKE in our sql syntax which should fit best.

**Comment by oparoz** [comment]
https://github.com/owncloud/core/pull/14947 using ILIKE

**Comment by MorrisJobke** [comment]
fixed with #14947
